packages/material-ui-csskit/src/responsiveTypography.js

```diff
@@ -0,0 +1,47 @@
+// import { allVariants } from '@material-ui/core/Typography';
+import fluidRange from './fluidRange';
+
+function remToPx(value) {
+  return Math.round(parseFloat(value) * 16);
+}
+
+export default function responsiveTypography(theme, { minFontSize, scale, ...other }) {
+  const output = theme;
+  output.typography = { ...theme.typography };
+
+  // TODO: use allVariants
```

Comment:
I think duplicating the variant names is not bad. They're hardcoded into material-design anyway. Having them static instead of dynamic is ok IMO.

Reply:
this and https://github.com/mui-org/material-ui/pull/14573#discussion_r257714475 were removed before the comments. Before this change, I was altering Typography on my side as were others and could use this (besides packages like these), and hardcodes would eventually generate issues. But I see the doc generator needed the comments that came with it. I'll try to do tests that check if hardcodes are out of date instead!

Reply:
> I was altering Typography on my side

That is in general not supported for any of our components. If you have a use case that is currently not covered you should open an issue. If you change internals of the components and we don't know about these use cases we might introduce breaking changes into your codebase at any time.

Reply:
thanks @eps1lon, I understand. It was just configure font styles in all variants and was adding this const to your public API to help (like this example), as otherwise changes to variant names bring breaking changes. Although I found you have a [`allVariants` option](https://github.com/mui-org/material-ui/pull/11802/files) that addresses this need exactly! But I think it's not on the docs. I'll look into this and open a separate PR if needed.